src/SourceBuild/Arcade/eng/common/templates/job/source-build-build-tarball.yml

```
@@ -43,27 +43,44 @@ jobs:
     - name: additionalBuildArgs
       value: --with-sdk /tarball/.dotnet
   - name: tarballDir
-    value: $(Build.StagingDirectory)/tarball
+    ${{ if eq(parameters.installerBuildResourceId, '') }}:
+      value: $(Build.SourcesDirectory)
+    ${{ else }}:
+      value: $(Build.StagingDirectory)/tarball
+  - name: installerSrcDir
+    ${{ if eq(parameters.installerBuildResourceId, '') }}:
+      value: $(Build.SourcesDirectory)/src/installer
+    ${{ else }}:
+      value: $(Build.SourcesDirectory)
   workspace:
     clean: all
 
   steps:
   - checkout: self
     clean: true
 
-  - ${{ if ne(variables['System.TeamProject'], 'public') }}:
-    - task: Bash@3
-      displayName: Setup Private Feeds Credentials
-      inputs:
-        filePath: $(Build.SourcesDirectory)/eng/common/SetupNugetSources.sh
-        arguments: $(Build.SourcesDirectory)/NuGet.config $Token
-      env:
-        Token: $(dn-bot-dnceng-artifact-feeds-rw)
+  - ${{ if ne(parameters.installerBuildResourceId, '') }}:
+    - download: ${{ parameters.installerBuildResourceId }}
+      artifact: BlobArtifacts
+      patterns: '**/dotnet-sdk-source*.tar.gz'
+      displayName: Download Source Tarball
+
+    - script: |
+        set -x
 
-  - download: ${{ parameters.installerBuildResourceId }}
-    artifact: BlobArtifacts
-    patterns: '**/dotnet-sdk-source*.tar.gz'
-    displayName: Download Source Tarball
+        resourceIdPathSegment=
+        if [[ '${{ parameters.installerBuildResourceId }}' != 'current' ]]; then
+          resourceIdPathSegment='${{ parameters.installerBuildResourceId }}/'
+        fi
+
+        mkdir -p "$(tarballDir)"
+        tarballFilePath="$(PIPELINE.WORKSPACE)/${resourceIdPathSegment}BlobArtifacts/dotnet-sdk-source*.tar.gz"
+        eval tar -ozxf "$tarballFilePath" -C "$(tarballDir)"
+        eval rm -f "$tarballFilePath"
+      displayName: Extract Tarball
+
+  - script: cp $(installerSrcDir)/NuGet.config $(tarballDir)/test/Microsoft.DotNet.SourceBuild.SmokeTests/assets/online.NuGet.Config
+    displayName: Copy NuGet Config for smoke tests
```

Shorter names are nice to avoid truncation in the AzDO UI.

```suggestion
    displayName: Copy Test NuGet Config
```